lib/ansible/galaxy/api.py

```diff
@@ -696,17 +697,29 @@ def wait_import_task(self, task_id, timeout=0):
         start = time.time()
         wait = 2
 
+        token_expired = False
         while timeout == 0 or (time.time() - start) < timeout:
+            if isinstance(self.token, KeycloakToken) and self.token.expired:
+                display.warning("Authentication failed, refreshing token to try again.")
+                self.token._token = None
```

Comment:
With https://github.com/ansible/ansible/pull/79145/files/0c2d16d6eaa97b5e1dc95cbddb199acac5aa0d74#r1013488315, it won't be necessary to mutate the private attributes of the token object.